pkg/machinepool/machinepool_test.go

```diff
@@ -225,6 +227,49 @@ var _ = Describe("Machinepool and nodepool", func() {
 				),
 			)
 		})
+		It("should return an error if both `subnet` and `availability-zone` flags are set", func() {
+			cmd := &cobra.Command{}
+			cmd.Flags().Bool("availability-zone", true, "")
+			cmd.Flags().Bool("subnet", true, "")
+
+			clusterKey := "test-cluster-key"
+			clusterBuilder := cmv1.NewCluster().ID("test").State(cmv1.ClusterStateReady)
+			cluster, err := clusterBuilder.Build()
```

Comment:
Can we update this to use the MockCluster func :pray: 
```suggestion
			cluster := MockCluster(func(c *cmv1.ClusterBuilder) {
				c.State(cmv1.ClusterStateReady)
				b := cmv1.HypershiftBuilder{}
				b.Enabled(true)
				c.Hypershift(&b)
			})
```